roles/mirror_images/tasks/mirror-images.yml

```diff
@@ -2,8 +2,9 @@
 - name: "Mirror image"
   vars:
     _image_tag: "{{ image | regex_search('(?<=sha256)(.{12})') | default('', True) }}"
-    _image_name: "{{ '/'.join(image.split('@')[0].split('/')[1:]) }}"
-    _tgt_image_location: "{{ mi_registry }}/{{ _image_name }}{%- if _image_tag | length %}{{ _image_tag }}{% endif %}"
+    _image_name: "{{ '/'.join(image.split('@')[0].split('/')[-1:]) }}"
+    _image_org: "{{ '/'.join(image.split('@')[0].split('/')[1:-1]) }}"
+    _tgt_image_location: "{{ mi_registry }}/{{ (mi_dst_org | length) | ternary(mi_dst_org, _image_org) }}/{{ _image_name }}{%- if _image_tag | length %}{{ _image_tag }}{% endif %}"
```

Comment:
`mi_dst_org` needs a default value otherwise it may fail

It would be nice if you split the org building and the image building to make it easier to read, for example

```yaml
_image_org_path: "{{ (mi_dst_org | length) | ternary(mi_dst_org, _image_org) }}"
_image_img_path: "{ _image_name }}{%- if _image_tag | length %}{{ _image_tag }}{% endif %}"
 _tgt_image_location: "{{ mi_registry }}/{{  _image_org_path }}/{{ _image_img_path }}"

Reply:
good idea, done